src/testdrive/src/action/kafka/verify.rs

```diff
@@ -261,10 +259,10 @@ impl Action for VerifyAction {
                 let mut actual_messages = vec![];
                 for (key, value) in actual_bytes {
                     let key_datum = match key {
-                        None => None,
-                        Some(bytes) => {
+                        Some(bytes) if *has_key => {
                             Some(serde_json::from_slice(&bytes).context("decoding json")?)
                         }
+                        _ => None,
```

Comment:
bit of a nit: can we make this explicit instead of `_`? It's pretty easy to get burned by the wildcard match arm

Reply:
Just to confirm: this line becomes `Some(_) | None => None`... still preferable? (Normally I agree but this one felt pretty weird.)

Or I guess I could rewrite using filter etc. on the option if that seems nicer...

Reply:
my take is weird but preferable, yeah -- or something like:
```
let key_datum = if has_key {
    match key {
     Some(bytes) => Some(serde_json...),
     None => None
   }
} else {
  None
}   
```

Reply:
cool - i'll go with that!